### PR TITLE
Support for Docker manifests base images add-on build

### PR DESCRIPTION
**Pull request by frenck**
## Proposed change

Since we support manifests now for add-on images, this PR set in motion to use a single base image for all architectures using a manifest, by allowing `build_from` to be a string (instead of mapping for each architecture).

So instead of doing this:

```yaml
# build.yaml
build_from:
  aarch64: ghcr.io/hassio-addons/base:12.2.0
  amd64: ghcr.io/hassio-addons/base:12.2.0
  armhf: ghcr.io/hassio-addons/base:12.2.0
  armv7: ghcr.io/hassio-addons/base:12.2.0
  i386: ghcr.io/hassio-addons/base:12.2.0
```

This PR will also allow for this:

```yaml
# build.yaml
build_from: ghcr.io/hassio-addons/base:12.2.0
```

Needs:
- [ ] Documentation adjustments
- [ ] Support for the builder to handle this
- [ ] Adjustment to add-on linter GitHub Action

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:
- Link to cli pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

**Review by pvizeli — COMMENTED** [review]
on `get_docker_args` we should set the right platform as we do on docker pull

**Comment by frenck** [comment]
> on get_docker_args we should set the right platform as we do on docker pull

It does already? Or am missing something?

**Comment by pvizeli** [comment]
If we do that, we can merge the PR

**Comment by mdegat01** [comment]
@frenck @pvizeli FYI I rebased this, added the platform to `get_docker_args` and added a couple tests so I think this is ready now if you both want to take a look.

**Comment by frenck** [comment]
Oh shit totally forgot about this one 🙈

File: tests/addons/test_config.py [hunk 1]
```diff
@@ -241,3 +248,7 @@ def test_watchdog_url():
     ):
         config["watchdog"] = test_options
         assert vd.SCHEMA_ADDON_CONFIG(config)
+
+
+async def test_addon_build_from():
+    """Test build per arch or manifest approach."""
```
Comment (frenck) [inline tests/addons/test_config.py:254]: empty test?

Comment (mdegat01) [reply tests/addons/test_config.py:254]: oh yea whoops, I started that then found the already existing one to test the normal `build_from`. Thanks

**Comment by frenck** [comment]
Looks good to me, however, I cannot approve as the PR author

**Review by pvizeli — APPROVED** [review]

**Pull request merged**